app/models/journaled/bulk_delivery.rb

```diff
@@ -40,6 +41,17 @@ def kinesis_client
     Journaled::KinesisClient.generate
   end
 
+  def reenqueue_failed_records!(response) # rubocop:disable Metrics/AbcSize
+    records_with_responses = records.zip(response.records)
+    errored_records = records_with_responses.select { |_record, resp| resp.error_code.present? }.map(&:first)
+
+    failed_record_count = response.failed_record_count || 0
+    raise 'FailedRecordCount differs from count of records that have errors' unless errored_records.count == failed_record_count
+    raise 'ALL Records failed to be added to the Kinesis steam' if errored_records.count == records.count
+
+    Delayed::Job.enqueue self.class.new(records: errored_records, app_name: app_name)
```

Comment:
This will NOT carry any of the enqueue opts from the current job to the next one, cause without larger changes that is hard to do in DJ

I also don't want to raise and fail this job since some of the records DID go through and we'd have more dupes to deal with, it also wouldn't help as much to decrease the amount of records we are putting on the stream

Reply:
it looks like this is always enqueuing a new job, even if no records fail? am I reading that right?

Reply:
🤦‍♂  Nice catch, I forgot the if any check!